src/compose/parser.c

```diff
@@ -734,10 +734,10 @@ lhs_mod_list_tok: {
                          "right-hand side string is too long: "
                          "expected max: %d, got: %d; skipping line",
                          (int)sizeof(production.string) - 1,
-                         (int)val.string.len);
+                         (int)val.string.len - 1);
             goto skip;
         }
-        strcpy(production.string, val.string.str);
+        strncpy(production.string, val.string.str, val.string.len);
```

Comment:
strncpy does useless padding of the destination string. And if it's not error checked then it's still not good, unless silent truncation is really the desired outcome. IMO it's a bad function so I'm not a huge fan of this commit.

strcpy is safe here since we check the sizes, but if want to avoid it then memcpy would be better.